server/aesdsocket.c

```diff
@@ -51,20 +77,98 @@ void socket_connect(void);
  */
 void signal_handler(int signal_no)
 {
+
 	if (signal_no == SIGINT || signal_no == SIGTERM || signal_no == SIGKILL)
 	{
 		printf("signal detected to exit\n");
 		syslog(LOG_DEBUG, "Caught the signal, exiting...");
-		unlink(file_data);	 // delete the file
-		free(output_buffer); // free output and send buffer
-		free(send_buffer);
-		close(accept_fd); // close fds
+		shutdown(socket_fd, SHUT_RDWR);
+		process_flag = true;
+		unlink(file_data);
+		close(accept_fd);
 		close(socket_fd);
 	}
-	exit(EXIT_SUCCESS);
+	_exit(0);
```

Comment:
@ayswariya088 I think you are missing cleanup in this code path.  You should avoid exit here and use a flag you set here to clean up from your main loop instead.